packages/react-scripts/scripts/start.js

```diff
@@ -51,6 +51,19 @@ if (!checkRequiredFiles([paths.appHtml, paths.appIndexJs])) {
 const DEFAULT_PORT = parseInt(process.env.PORT, 10) || 3000;
 const HOST = process.env.HOST || '0.0.0.0';
 
+if (process.env.HOST) {
+  console.log(
+    chalk.yellow(
+      "Attempting to bind to $HOST because it was specified. If this was unintentional, check that you haven't mistakenly set it in your shell."
```

Comment:
Shouldn't we inline the `HOST` value in the message?

Also it's not 100% clear what "$HOST" is. (For example windows uses a different syntax for environment variables.) It is better to explicitly say "environment variable".

Reply:
What if it was something more like "Attempting to bind to HOST environment variable somehostname.example.com. If this was..."